pkg/network/ebpf/c/tls.h

```diff
@@ -0,0 +1,129 @@
+#ifndef __TLS_H
+#define __TLS_H
+
+#include "tracer.h"
+#include "tls-types.h"
+#include "tls-maps.h"
+#include "ip.h"
+#include "http.h"
+
+#include <uapi/linux/ptrace.h>
+
+// TLS classification generate an HTTPStats event
+// that will include the tag and the latency of the connection
+//
+// Latency connection = time from the 1st packet (SYN) to the 1st payload packet (TLS_APP)
+static __always_inline void http_enqueue_with_tags(skb_info_t *skb_info, u64 tags, u64 started, u64 classified) {
+    http_transaction_t new_entry = { 0 };
+    new_entry.owned_by_src_port = skb_info->tup.sport;
+    bpf_map_update_elem(&http_in_flight, &skb_info->tup, &new_entry, BPF_NOEXIST);
+    http_transaction_t *http = bpf_map_lookup_elem(&http_in_flight, &skb_info->tup);
+    if (!http) {
+        return;
+    }
+    int have_tag = http->tags > 0;
+    if (!have_tag) {
+        http->request_started = started;
+        http->response_last_seen = classified;
+        http->response_status_code = 200;
+    }
+    http->tags |= tags;
+    if (!have_tag) {
+        http_enqueue(http, &skb_info->tup);
```

Comment:
so one bit of context that I'm missing here: are we using an `http_transaction_t` as a hack to send TLS information to userspace or does a HTTP request actually get sent during a TLS handshake? because if this is a hack that raises several questions to me:
* Is TLS classification supposed to work when HTTP monitoring is disabled?
* Should we have a special handling of these "fake" HTTP transactions on userspace?
* What will happen to L7 monitoring and USM monitoring product when they receive these fake HTTP transactions? (cc @brycekahle )
* I've mentioned this in my previous comment, but these are currently invalid as they're missing a path and a method;

Reply:
Also, instead of flushing things to userspace for protocol classification it may be more interesting to:
1) Piggy back on the `socket/dns_filter` program instead, as this program is much less bloated than HTTP. (we could just give it a more general purpose name;
2) Consider sharing the `maps/conn_stats` map and adding the tags directly to the map entry;

(and by the way, these changes I'm suggesting are mostly about where to hook your code, but they shouldn't require a lot of code changes 😅 )

Reply:
> so one bit of context that I'm missing here: are we using an `http_transaction_t` as a hack to send TLS information to userspace or does a HTTP request actually get sent during a TLS handshake? because if this is a hack that raises several questions to me:

You have good point, I will try to answer them, but I think we should have quick meeting about with @brycekahle 

> * Is TLS classification supposed to work when HTTP monitoring is disabled?

There are no real http request during TLS handshake, this hack/patch will enqueue an httpStats to send the tag :
 * we can drop this mechanism (send the TLS handshake time), we must enable http{,s} monitoring to support it. 

> * Should we have a special handling of these "fake" HTTP transactions on userspace?
> * What will happen to L7 monitoring and USM monitoring product when they receive these fake HTTP transactions? (cc @brycekahle )
> * I've mentioned this in my previous comment, but these are currently invalid as they're missing a path and a method;

Fake or not, I'm sure today how to present it to the backend (and as today it's not required), as it's per protocol metadata and could be different, I talked a little with @heyronhay but it's broader. I looking to add that in the "notes/features on protocols classification" document

Reply:
> Also, instead of flushing things to userspace for protocol classification it may be more interesting to:
> 
> 1. Piggy back on the `socket/dns_filter` program instead, as this program is much less bloated than HTTP. (we could just give it a more general-purpose name);
> 2. Consider sharing the `maps/conn_stats` map and adding the tags directly to the map entry;
> 
> (and by the way, these changes I'm suggesting are mostly about where to hook your code, but they shouldn't require a lot of code changes 😅 )

Totally agree we should share a connection maps with some features like is_classified, drop_packets, ... 
I was plan to work on that and use tail_call.
dns or http will not make a big difference as tls is in tail_call() so have a limited impact on instructions